pkg/acs/calacs/acs2d/dodark.c

```diff
@@ -53,152 +52,166 @@
   }
   
-	/* Compute correct extension version number to extract from
-   reference image to correspond to CHIP in science data.
-   */
-	if (acs2d->pctecorr == PERFORM) {
-    if (DetCCDChip (acs2d->darkcte.name, acs2d->chip, acs2d->nimsets, &extver) )
-      return (status);
+  /* Compute correct extension version number to extract from
+     reference image to correspond to CHIP in science data.
+  */
+  if (acs2d->pctecorr == PERFORM) {
+     if (DetCCDChip (acs2d->darkcte.name, acs2d->chip, acs2d->nimsets, &extver) )
+        return (status);
   } else {
-    if (DetCCDChip (acs2d->dark.name, acs2d->chip, acs2d->nimsets, &extver) )
-      return (status);
+     if (DetCCDChip (acs2d->dark.name, acs2d->chip, acs2d->nimsets, &extver) )
+        return (status);
   }
 	
-	if (acs2d->verbose) {
-		sprintf(MsgText,"Performing dark subtraction on chip %d in imset %d",acs2d->chip, extver);
-		trlmessage(MsgText);
-	}
+  if (acs2d->verbose) {
+     sprintf(MsgText,"Performing dark subtraction on chip %d in imset %d",acs2d->chip, extver);
+     trlmessage(MsgText);
+  }
   
-	/* Get the dark image data. */
+  /* Initialze and get the dark image data */
+  initSingleGroup (&y);
+
   if (acs2d->pctecorr == PERFORM) {
-    openSingleGroupLine (acs2d->darkcte.name, extver, &y);
+      getSingleGroup (acs2d->darkcte.name, extver, &y);
   } else {
-    openSingleGroupLine (acs2d->dark.name, extver, &y);
+      getSingleGroup (acs2d->dark.name, extver, &y);
+  }
+
+  if (hstio_err()) {
+     freeSingleGroup (&y);
+     return (status = OPEN_FAILED);
+  }
+
+  /* Compare binning of science image and reference image; get same_size 
+     and high_res flags, and get info about binning and offset for 
+     use by bin2d.
+  */
+  if (FindBin (x, &y, &same_size, &rx, &ry, &x0, &y0)) {
+     freeSingleGroup (&y);
+     return (status);
   }
-	if (hstio_err())
-    return (status = OPEN_FAILED);
-  
-  
-	/* Compare binning of science image and reference image;
-   get same_size and high_res flags, and get info about
-   binning and offset for use by bin2d.
-   */
-	if (FindLine (x, &y, &same_size, &rx, &ry, &x0, &y0))
-    return (status);
   
   if (rx != 1 || ry != 1) {
-		sprintf(MsgText,"Reference image and input are not binned to the same pixel size!");
-		trlmessage(MsgText);
+     sprintf(MsgText,"Reference image and input are not binned to the same pixel size!");
+     trlmessage(MsgText);
+  }
+  if (acs2d->verbose){
+     sprintf(MsgText,"Image has an offset of %d,%d",x0,y0);
+     trlmessage(MsgText);
   }
-	if (acs2d->verbose){
-		sprintf(MsgText,"Image has an offset of %d,%d",x0,y0);
-		trlmessage(MsgText);
-	}
-  
-	mean = 0.0;
-  weight = 0.0;
-  
-	/* Bin the dark image down to the actual size of x. */
   
-	initSingleGroupLine (&z);
-	allocSingleGroupLine (&z, x->sci.data.nx);
-	for (i=0, j=y0; i < scilines; i++,j++) { 
+  /* Trim the dark image (y->z) down to the actual size of the science image (x). */
+  initSingleGroup (&z);
```

Comment:
I would move the dec for ``z`` to here or the ``initSingleGroup`` to just after the dec. Pref the former, either way, the uninitialized gap needs to be closed.

Reply:
Updated for y and z.